src/main/java/io/openliberty/tools/intellij/util/DebugModeHandler.java

```diff
@@ -0,0 +1,276 @@
+     */
+    private Path getServerEnvPath(LibertyModule libertyModule) throws Exception {
+        String projectPath = libertyModule.getBuildFile().getParent().getPath();
+        Path basePath = null;
+        if (libertyModule.getProjectType().equals(Constants.LIBERTY_MAVEN_PROJECT)) {
+            basePath = Paths.get(projectPath, "target", "liberty", "wlp", "usr", "servers");
+        } else if (libertyModule.getProjectType().equals(Constants.LIBERTY_GRADLE_PROJECT)) {
+            basePath = Paths.get(projectPath, "build", "wlp", "usr", "servers");
+        } else {
+            throw new Exception(String.format("Unexpected project build type: %s. Liberty module %s does not appear to be a Maven or Gradle built project",
+                    libertyModule.getProjectType(), libertyModule.getName()));
+        }
+
+        // Make sure the base path exists. If not return null.
+        File basePathFile = new File(basePath.toString());
+        if (!basePathFile.exists()) {
+            return null;
+        }
+
+        try (Stream<Path> matchedStream = Files.find(basePath, 2, (path, basicFileAttribute) -> {
+            if (basicFileAttribute.isRegularFile()) {
+                return path.getFileName().toString().equalsIgnoreCase("server.env");
+            }
+            return false;
+        });) {
+            List<Path> matchedPaths = matchedStream.collect(Collectors.toList());
+            int numberOfFilesFound = matchedPaths.size();
+
+            if (numberOfFilesFound != 1) {
+                if (numberOfFilesFound == 0) {
+                    LOGGER.trace(String.format("Unable to find the server.env file for project %s", libertyModule.getName()));
+                    return null;
+                } else {
+                    throw new Exception(String.format("More than one server.env files were found for project %s. Unable to determine the server.env file to use", libertyModule.getName()));
+                }
+            }
+            return matchedPaths.get(0);
+        }
+    }
+
+    /**
+     * Returns the last debug port entry in server.env. Null if not found.
+     *
+     * @param serverEnv The server.env file object.
+     *
+     * @return The last debug port entry in server.env. Null if not found.]]
+     *
+     * @throws Exception
+     */
+    private String readDebugPortFromServerEnv(File serverEnv) throws Exception {
+        String port = null;
+
+        if (serverEnv.exists()) {
+            try (BufferedReader reader = new BufferedReader(new FileReader(serverEnv))) {
+                String line = null;
+                String lastEntry = null;
+                while ((line = reader.readLine()) != null) {
+                    if (line.contains(WLP_ENV_DEBUG_ADDRESS)) {
+                        lastEntry = line;
+                    }
+                }
+                if (lastEntry != null) {
+                    String[] parts = lastEntry.split("=");
+                    port = parts[1].trim();
```

Comment:
Do we have a guarantee that `lastEntry` will always have value? Otherwise, `parts[1]` can be null. 
I see that the method is prepared to throw Exception, but maybe would be better to avoid the `
NullPointerException`.

Reply:
We do the null check on line 267, were you thinking we need additional checks?

Reply:
I see... I think I expressed myself in the wrong way... I would say that if I understood it right, `String[] parts = lastEntry.split("=");` potentially could have just one array position, and then trying to access `parts[1].trim();` could cause the `NullPointerException`.

Reply:
I see what you mean, good point! I've added a length check